api/models/site.js

```diff
@@ -154,9 +157,15 @@ module.exports = (sequelize, DataTypes) => {
     },
     previewConfig: {
       type: DataTypes.STRING,
+      validate: {
+        isValidYaml,
+      },
     },
     demoConfig: {
       type: DataTypes.STRING,
+      validate: {
+        isValidYaml,
```

Comment:
empty strings will validate successfully, so no need to add an explicit `isEmpty` check